data.sql

```diff
@@ -12,17 +12,39 @@ INSERT INTO species (name) VALUES
     ('Pokemon'),
     ('Digimon');
 
--- Modify animals based on name and assign species_id
-UPDATE animals SET species_id = (CASE
-    WHEN name LIKE '%mon' THEN (SELECT id FROM species WHERE name = 'Digimon')
-    ELSE (SELECT id FROM species WHERE name = 'Pokemon')
-END);
+-- Insert data into the 'vets' table
+INSERT INTO vets (name, age, date_of_graduation) VALUES
+    ('Vet William Tatcher', 45, '2000-04-23'),
+    ('Vet Maisy Smith', 26, '2019-01-17'),
+    ('Vet Stephanie Mendez', 64, '1981-05-04'),
+    ('Vet Jack Harkness', 38, '2008-06-08');
 
--- Modify animals and assign owner_id
-UPDATE animals SET owner_id = (CASE
-    WHEN name IN ('Agumon') THEN (SELECT id FROM owners WHERE full_name = 'Sam Smith')
-    WHEN name IN ('Gabumon', 'Pikachu') THEN (SELECT id FROM owners WHERE full_name = 'Jennifer Orwell')
-    WHEN name IN ('Devimon', 'Plantmon') THEN (SELECT id FROM owners WHERE full_name = 'Bob')
-    WHEN name IN ('Charmander', 'Squirtle', 'Blossom') THEN (SELECT id FROM owners WHERE full_name = 'Melody Pond')
-    WHEN name IN ('Angemon', 'Boarmon') THEN (SELECT id FROM owners WHERE full_name = 'Dean Winchester')
-END);
+-- Insert data into the 'specializations' table
+INSERT INTO specializations (vet_id, species_name) VALUES
+    (1, 'Pokemon'),
+    (3, 'Digimon'),
+    (3, 'Pokemon'),
+    (4, 'Digimon');
+
+-- Insert data into the 'visits' table
+INSERT INTO visits (animal_id, vet_id, visit_date) VALUES
+    (1, 1, '2020-05-24'),
+    (1, 3, '2020-07-22'),
+    (2, 4, '2021-02-02'),
+    (3, 2, '2020-01-05'),
+    (3, 2, '2020-03-08'),
+    (3, 2, '2020-05-14'),
+    (4, 3, '2021-05-04'),
+    (5, 4, '2021-02-24'),
+    (6, 2, '2019-12-21'),
+    (6, 1, '2020-08-10'),
+    (6, 2, '2021-04-07'),
+    (7, 3, '2019-09-29'),
+    (8, 4, '2020-10-03'),
+    (8, 4, '2020-11-04'),
+    (9, 2, '2019-01-24'),
+    (9, 2, '2019-05-15'),
+    (9, 2, '2020-02-27'),
+    (9, 2, '2020-08-03'),
+    (10, 3, '2020-05-24'),
+    (11, 1, '2021-01-11');
```

Comment:
👍 You're on the right track with your implementation of the `visits` table! However, there's room for improvement to ensure universality across different databases. Here's a suggestion:

- [ ] Instead of hardcoding the `vet_id` and `animal_id` values in the `visits` table, consider dynamically retrieving these values based on the vet's and animal's names. For example, you can first query the `vets` table to find the `id` of a vet with the name 'William Tatcher' and then query the `animals` table to find the `id` of an animal with a specific name. Once you have these `id` values, you can populate them in the `visits` table. This approach ensures that your queries run universally across different databases without errors.

example:
```
INSERT INTO visits (vet_id, animal_id, visit_date)
VALUES (
    (SELECT id FROM vets WHERE name = 'William Tatcher'),
    (SELECT id FROM animals WHERE name = 'Buddy'),
    '2023-08-31'
);

```

By making this adjustment, your code will be more adaptable and robust, providing consistent results regardless of the database configuration. Keep up the great work! 👍🔍🌐